integration-tests/features/steps/stack_analyses_v2.py

```diff
@@ -0,0 +1,376 @@
+    # Ecosystem is mandatory
+    assert ecosystem is not None
+
+    # Convert token text into a valid bool
+    with_user_key = parse_token_clause(token)
+
+    # Convert valid clause to bool
+    is_valid = parse_valid_clause(valid)
+
+    # Send SA request
+    sav2_post_request(context, ecosystem, manifest, with_user_key, is_valid)
+
+
+@when('I wait for stack analysis v2 to finish {token} user key')
+def sav2_wait_for_completion(context, token='without'):
+    """Try to wait for the stack analysis to be finished.
+
+    This step assumes that stack analysis has been started previously and
+    thus that the request id is known
+
+    Current API implementation returns just three HTTP codes:
+    200 OK : analysis is already finished
+    202 Accepted: analysis is started or is in progress (or other state!)
+    401 UNAUTHORIZED : missing or improper user key
+    408 Timeout: Request timeout.
+    """
+    context.duration = None
+    start_time = time.time()
+
+    timeout = context.stack_analysis_timeout  # in seconds
+    sleep_amount = 10  # we don't have to overload the API with too many calls
+    with_user_key = parse_token_clause(token)
+
+    id = context.response.json().get('id')
+    context.stack_analysis_id = id
+    print('SA V2 Request id: {}'.format(id))
+
+    url = urljoin(sav2_get_endpoint(context), id)
+    print('Get API url: {}'.format(url))
+
+    for _ in range(timeout // sleep_amount):
+        if with_user_key:
+            params = {'user_key': context.three_scale_preview_user_key}
+            context.response = requests.get(url, params=params)
+        else:
+            context.response = requests.get(url)
+        status_code = context.response.status_code
+        # print('status_code: {} response: {}'.format(status_code, context.response.text))
+        print('status_code: {}'.format(status_code))
+        if status_code == 200:
+            break
+        # 401 code should be checked later
+        elif status_code == 401:
+            break
+        elif status_code != 202:
+            raise Exception('Bad HTTP status code {c}'.format(c=status_code))
+        time.sleep(sleep_amount)
+    else:
+        raise Exception('Timeout waiting for the stack analysis results')
+
+    end_time = time.time()
+    # compute the duration
+    # plase note that duration==None in case of any errors (which is to be expected)
+    context.duration = end_time - start_time
+
+
+@then('I should find the external request id equals to id returned by '
+      'stack analysis v2 post request')
+def sav2_check_get_request_id(context):
+    """Check the ID of stack analysis."""
+    previous_id = context.stack_analysis_id
+    assert previous_id is not None
+
+    json_data = get_json_data(context)
+
+    id_name = 'external_request_id'
+    check_attribute_presence(json_data, id_name)
+    request_id = json_data[id_name]
+    print('Post id: {} Get id: {}'.format(previous_id, request_id))
```

Comment:
remove print

Reply:
Same justification as before. Also there are hardly 4-5 prints in entire code. These prints were very useful when a test case fails and they are displayed only in case of failed test case, otherwise they are not visible on console logs. I hope this information helps.